application.cpp

```diff
@@ -445,14 +462,18 @@ void application::exec() {
       (void)work;
       bool more = true;
       while( more || io_serv->run_one() ) {
+         if (my->_is_quiting)
```

Comment:
I'm not initially seeing why this is required -- `run_one()` and `poll_one()` are both documented to do nothing should the io_context be stopped.

Reply:
There is the edge case where `run_one()` checks whether the `io_context` is stopped, and it is not, but while processing the action and posting the task on the priority_queue, another thread calls `application::quit()` stopping the io_context. As you suggested above, it is appropriate to not process tasks anymore at this point.

Reply:
`more` is true if there are items in the priority queue. So the current code drains the queue after `quit()`. If we want to change that behavior to exit without executing queued items, which I believe we do, then we need something like `my->_is_quiting` check.

Reply:
yup, thanks